File: 2023/Leetcode/problem122.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let totalProfit = 0;

    for (let i = 0; i < prices.length - 1; i++) {
        const buyPrice = prices[i];
        const sellPrice = prices[i + 1];
        let potentialProfit = sellPrice - buyPrice;

        if (potentialProfit > 0) {
            totalProfit += potentialProfit;
        }
    }
    
    return totalProfit
};

File: 2023/Leetcode/problem122.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        result = 0
        for i in range(len(prices) - 1):
            buyprice = prices[i]
            sellprice = prices[i + 1]

            profit = sellprice - buyprice
            if profit > 0:
                result += profit
        
        return result
